### PR TITLE
Move code around

### PR DESCRIPTION
**Pull request by millerdev**
- Group error handling functions in core.js
- Move Mug class to top of mugs.js

This was cherry-picked from the copy-paste-dev (e0fef536473f678480d5206d628f3d2e15a78b0a), and it should make merges easier.

@emord @orangejenny 
cc @benrudolph

### CODE REVIEW

File: src/mugs.js [hunk 1]
```diff
@@ -444,176 +601,6 @@ define([
...
-        setItextID: function (val) {
-            var labelItext = this.p.labelItext;
-            if (labelItext) {
-                labelItext.id = val;
-            }
-        },
-        
-        getItext: function () {
-            return this.p.labelItext;
-        },
```
Comment (millerdev) [inline src/mugs.js]: The above two (unused) Itext functions were removed during the code move. Should have at least done that as a separate commit...

**Comment by orangejenny** [comment]
ran tests locally
